packages/pds/src/feed-gen/whats-hot.ts

```diff
@@ -0,0 +1,109 @@
+import AppContext from '../context'
+import { DAY, NotEmptyArray } from '@atproto/common'
+import { QueryParams as SkeletonParams } from '@atproto/api/src/client/types/app/bsky/feed/getFeedSkeleton'
+import { AlgoHandler, AlgoResponse } from './types'
+import { sql } from 'kysely'
+import { InvalidRequestError } from '@atproto/xrpc-server'
+
+const NO_WHATS_HOT_LABELS: NotEmptyArray<string> = [
+  '!no-promote',
+  'corpse',
+  'self-harm',
+  'porn',
+  'sexual',
+  'nudity',
+  'underwear',
+]
+
+const handler: AlgoHandler = async (
+  ctx: AppContext,
+  params: SkeletonParams,
+  requester: string,
+): Promise<AlgoResponse> => {
+  if (ctx.db.dialect === 'sqlite') {
+    throw new Error('what-hot algo not available in sqlite')
+  }
+
+  const { limit = 50, cursor } = params
+  const accountService = ctx.services.account(ctx.db)
+  const feedService = ctx.services.appView.feed(ctx.db)
+  const graphService = ctx.services.appView.graph(ctx.db)
+
+  const { ref } = ctx.db.db.dynamic
+
+  // From: https://medium.com/hacking-and-gonzo/how-hacker-news-ranking-algorithm-works-1d9b0cf2c08d
+  // Score = (P-1) / (T+2)^G
+  // where,
+  // P = points of an item (and -1 is to negate submitters vote)
+  // T = time since submission (in hours)
+  // G = Gravity, defaults to 1.8 in news.arc
+
+  const computeScore = sql<number>`${ref(
```

Comment:
not sure how crazy it is to compute this for every post in the last 24 hrs 😅 

we'll need to make it smarter over time